templates/github/.ci/scripts/changelog.py.j2

```diff
@@ -0,0 +1,75 @@
+import re
+import os
+import requests
+from packaging.version import Version
+from git import Repo
+
+resp = requests.get("https://api.github.com/repos/pulp/{{ plugin_snake }}/branches")
+branches = [b["name"] for b in resp.json() if re.search(r"^([0-9]+)\.([0-9]+)$", b["name"])]
+branches.sort(key=lambda ver: Version(ver), reverse=True)
+
+
+def get_changelog(branch):
+    """
+    Get changelog file for a given branch
+
+    """
+    return requests.get(
+        f"https://raw.githubusercontent.com/pulp/{{ plugin_snake }}/{branch}/CHANGES.rst"
+    ).text
+
+
+def get_changelog_releases(changelog):
+    """
+    Get changelog entries for all releases
+
+    """
+    versions = re.findall(r"([0-9]+)\.([0-9]+)\.([0-9]+) \(", changelog)
+    return {".".join(v) for v in versions}
```

Comment:
```suggestion
    Get all versions in changelog.

    """
    versions = re.findall(r"([0-9]+\.[0-9]+\.[0-9]+) \(", changelog)
    return {v[0] for v in versions}
```
Assuming version behaves like a list of lists.

Reply:
it returns  a list [X, Y, Z]
```
> /home/faguiard/project/pulpcore/.ci/scripts/changelog.py(29)get_changelog_releases()
-> return {".".join(v) for v in versions}
(Pdb) versions [0]
('3', '18', '1')
(Pdb) ".".join(versions[0])
'3.18.1'
(Pdb) 
```

Reply:
My point is that we don't need 3 capture groups for the individual numbers if we could also use one capture group for the version in its entirety.